core/includes/database/database.inc

```diff
@@ -1746,7 +1746,22 @@ abstract class Database {
     require_once BACKDROP_ROOT . '/core/includes/database/' . $driver . '/database.inc';
 
     /* @var DatabaseConnection $new_connection */
-    $new_connection = new $driver_class(self::$databaseInfo[$key][$target]);
+    try {
+      $new_connection = new $driver_class(self::$databaseInfo[$key][$target]);
+    }
+    catch (PDOException $e) {
+      // If the password is incorrect, try re-encoding the password and signing
+      // in again. This can occur because the password string needs to be
+      // urlencoded for PHP 7 and higher, but Backdrop versions prior to 1.9 did
+      // not require this encoding.
+      if ($e->getCode() === 1045) {
```

Comment:
add a reference to https://dev.mysql.com/doc/refman/5.7/en/error-messages-server.html#error_er_access_denied_error ? to explain the 1045